### PR TITLE
U/ejeffrey/fasttypes

### PR DESCRIPTION
**Pull request by ejeffrey**
This fixes the crash when no type tag matches and adds tests.

### CODE REVIEW

File: labrad/test/test_fasttypes.py [hunk 1]
```diff
@@ -8,14 +8,20 @@ def test_int():
...
     with pytest.raises(Exception):
         ft.flatten(4, 's')
 def test_list():
     assert all(T.unflatten(*ft.flatten([1,2,3])) == [1,2,3])
-    assert all(ft.unflatten(T.flatten([1,2,3])) == [1,2,3])
+    (data, tt) = T.flatten([1,2,3])
+    data, tt = str(data), str(tt)
+    assert all(ft.unflatten((data, tt)) == [1,2,3])
+    assert len(ft.flatten(['a', 'bc', 'def'])[0])==22
+def test_list_w_tag():
```
Comment (maffoo) [inline labrad/test/test_fasttypes.py]: style: add blank lines between top-level definitions (https://google-styleguide.googlecode.com/svn/trunk/pyguide.html?showone=Blank_Lines#Blank_Lines)

File: labrad/test/test_fasttypes.py [hunk 2]
```diff
@@ -8,14 +8,20 @@ def test_int():
...
+def test_wrong_type(): # Figure out the right exceptions to use here
     with pytest.raises(Exception):
         ft.flatten('abcd', 'i')
     with pytest.raises(Exception):
         ft.flatten(4, 's')
 def test_list():
     assert all(T.unflatten(*ft.flatten([1,2,3])) == [1,2,3])
-    assert all(ft.unflatten(T.flatten([1,2,3])) == [1,2,3])
+    (data, tt) = T.flatten([1,2,3])
+    data, tt = str(data), str(tt)
```
Comment (maffoo) [inline labrad/test/test_fasttypes.py]: why do you have to call `str` on these? what is the actual return type?

File: labrad/test/test_fasttypes.py [hunk 3]
```diff
@@ -8,14 +8,20 @@ def test_int():
...
+    assert all(ft.unflatten((data, tt)) == [1,2,3])
+    assert len(ft.flatten(['a', 'bc', 'def'])[0])==22
```
Comment (maffoo) [inline labrad/test/test_fasttypes.py]: nit: whitespace around operators

File: labrad/test/test_fasttypes.py (hunk 2 above)
Comment (ejeffrey) [inline labrad/test/test_fasttypes.py]: fasttypes only works with strings
types.py returns LRType objects and sometimes uses Buffer()'s for the data.

Comment (maffoo) [inline labrad/test/test_fasttypes.py]: ah, right, these are coming from `types.py` not `fasttypes`

**Comment by maffoo** [comment]
LGTM

**Comment by ejeffrey** [comment]
Fixed a memory leak austin pointed out.  I also found a few more memory
leaks and uninitialized memory use with valgrind that still need to be
fixed.  I want to get him to review this. @afowler.

**Comment by afowler** [comment]
Would love to review the changes, struggling to find them...

**Comment by ejeffrey** [comment]
If you follow the github link at the bottom of the email (Reply to this
email directly or view it on GitHub) you can see the history of comments on
the pull request.  Near the top there are 'tabs' for
Conversation, Commits, and Files changed.  Files changed shows the total
difference between my version and yours, while commits lets you look at
invidual commits.

**Comment by afowler** [comment]
Hi Evan,

So these extra memory issues have not been addressed yet?

> > I also found a few more memory
> > leaks and uninitialized memory use with valgrind that still need to be
> > fixed.

Assuming this is the case, LGTM.

**Pull request merged**